conans/test/optimize_conanfile_load_test.py

```diff
@@ -0,0 +1,27 @@
+import unittest
+
+from conans.test.utils.tools import TestClient
+
+
+class OptimizeConanFileLoadTest(unittest.TestCase):
+
+    def test_multiple_load(self):
+        client = TestClient()
+        conanfile = """from conans import ConanFile
+mycounter = 0
+class Pkg(ConanFile):
+    def configure(self):
+        global mycounter
+        mycounter += 1
+        self.output.info("My Counter %s" % mycounter)
+"""
+        client.save({"conanfile.py": conanfile})
+
+        client.run("create . Build/0.1@user/testing")
+
+        client.save({"conanfile.py": conanfile,
+                     "test_package/conanfile.py": conanfile + "    def test(self): pass",
+                     "myprofile": "[build_requires]\nBuild/0.1@user/testing"})
+
+        client.run("create . Pkg/0.1@user/testing -pr=myprofile")
+        self.assertIn("Build/0.1@user/testing: My Counter 2", client.out)
```

Comment:
Does the value of `mycounter` persist due to the fact that the test runs all at once or does it really cache the conanfile from one conan command to another? I think it checks the first case so this test might not be very representative

Reply:
The module is imported just once, as opposed as without the PR, that the module is really imported as a totally new module, for every node of the graph, and the "Build" package is 2 different nodes of the graph, because it is build-required by the 2 consumer conanfiles.

As now it is imported once, variables declared at the module level are shared between those 2 nodes of the graph.

Reply:
Yes, I understood the purpose. What I said is that this test might not be representative of that purpose

Reply:
Ok, I got it! 👍